human-dynamics-estimator/include/HumanDynamicsEstimator.h

```diff
@@ -1,29 +1,122 @@
 
-//
-//  HumanDynamicsEstimator.h
-//  HumanDynamicsEstimator
-//
-//  Created by Claudia Latella on 03/02/17.
-//  Copyright © 2017 Claudia Latella. All rights reserved.
 
 #ifndef HUMANDYNAMICSESTIMATOR_H
 #define HUMANDYNAMICSESTIMATOR_H
 
+#include <unordered_map>
+
 #include <yarp/os/RFModule.h>
+#include <yarp/os/BufferedPort.h>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/sig/Vector.h>
+
 #include <iDynTree/Estimation/BerdyHelper.h>
+#include <iDynTree/KinDynComputations.h>
+#include <iDynTree/Core/SparseMatrix.h>
+
+#include<Eigen/Dense>
+#include<Eigen/Sparse>
+#include<Eigen/Core>
+#include <Eigen/SparseCholesky>
+
+#include <thrifts/HumanDynamics.h>
+#include <thrifts/HumanState.h>
+#include <thrifts/HumanForces.h>
+
+
+
+namespace human
+{
+    class HumanDynamics;
+    class HumanState;
+    class HumanForces;
+
+}
 
 
 class HumanDynamicsEstimator:public yarp::os::RFModule {
 
 private:
-    std::string modelFilename;
-    int count;
+    int m_count;
+    
+    double m_period;
+    
+    //Port
+    yarp::os::BufferedPort<human::HumanState> m_humanJointConfiguration_port; 
+    
+    yarp::os::BufferedPort<human::HumanForces> m_humanForces_port;
+    
+    //buffered port:o from <human-dynamics-estimator> module
+    yarp::os::BufferedPort<human::HumanDynamics> m_outputPort;
     
+    //Berdy
+    iDynTree::BerdyHelper m_berdy;
+    
+    std::vector<std::string> m_linkName;
+    std::vector<std::string> m_jointName;
+    
+    std::unordered_map<std::string,int> m_y_map;
+    typedef std::unordered_map<std::string,iDynTree::IndexRange> BerdyOutputRangeMap;
+    typedef std::unordered_map<std::string,std::unordered_map<std::string,iDynTree::IndexRange> > BerdyOutputMap;
```

Comment:
This map is simply 
`typedef std::unordered_map<std::string, BerdyOutputRangeMap > > BerdyOutputMap;`